src/history/cam_hist_file.F90

```diff
@@ -0,0 +1,2162 @@
+
+   function config_filename(this) result(cfiles)
+      use cam_filenames,  only: interpret_filename_spec
+      use cam_abortutils, only: check_allocate
+      ! Dummy arguments
+      class(hist_file_t), intent(in) :: this
+      character(len=CL), allocatable :: cfiles(:)
+
+      character(len=1) :: accum_types(max_split_files)
+      integer :: file_idx, ierr
+      character(len=*), parameter :: subname = 'config_filename: '
+
+      accum_types(instantaneous_file_index) = 'i'
+      accum_types(accumulated_file_index)   = 'a'
+      allocate(cfiles(max_split_files), stat=ierr)
+      call check_allocate(ierr, subname, 'cfiles',             &
+           file=__FILE__, line=__LINE__-1)
+
+      do file_idx = 1, size(cfiles, 1)
+         cfiles(file_idx) = interpret_filename_spec(this%filename_spec, &
+           unit=this%volume, accum_type=accum_types(file_idx),          &
+           incomplete_ok=.false.)
+      end do
+
+   end function config_filename
+
+   ! ========================================================================
+
+   subroutine config_set_filenames(this)
+      use cam_filenames,  only: interpret_filename_spec
+      use cam_abortutils, only: check_allocate
+      ! Dummy argument
+      class(hist_file_t), intent(inout) :: this
+      
+      character(len=1) :: accum_types(max_split_files)
+      integer          :: file_idx, ierr
+      character(len=*), parameter :: subname = 'config_set_filenames: '
+
+      if (allocated(this%file_names)) then
+         return
+      end if
+      accum_types(instantaneous_file_index) = 'i'
+      accum_types(accumulated_file_index)   = 'a'
+      allocate(this%file_names(max_split_files), stat=ierr)
+      call check_allocate(ierr, subname, 'this%file_names',   &
+           file=__FILE__, line=__LINE__-1)
+      do file_idx = 1, size(this%file_names, 1)
+         this%file_names(file_idx) = interpret_filename_spec(this%filename_spec, &
+           unit=this%volume, accum_type=accum_types(file_idx),          &
+           incomplete_ok=.false.)
+      end do
+
+   end subroutine config_set_filenames
+
+   ! ========================================================================
+
+   subroutine config_set_last_month_written(this, last_month_written)
+      ! Dummy arguments
+      class(hist_file_t), intent(inout) :: this
+      integer,               intent(in) :: last_month_written
+
+      this%last_month_written = last_month_written
+
+   end subroutine config_set_last_month_written
+
+   ! ========================================================================
+
+   subroutine config_set_last_year_written(this, last_year_written)
+      ! Dummy arguments
+      class(hist_file_t), intent(inout) :: this
+      integer,               intent(in) :: last_year_written
+
+      this%last_year_written = last_year_written
+
+   end subroutine config_set_last_year_written
+
+   ! ========================================================================
```

Comment:
Because the setters don't return a value, I got stuck on how to achieve that. I guess I could return a dummy integer? Is there a way that I'm missing?